### ISSUE TITLE
Mapping and Aligning road network in Taiwan based on new Mapbox imagery and Strava

### ISSUE THREAD

**Issue by jothirnadh** [mapbox/mapping#233]
### Objective:

Add/ realign the roads in Taiwan using Mapbox satellite imagery and Strava

![taiwan_freedom_plaza_11-12-2007f](https://cloud.githubusercontent.com/assets/8401827/18703279/d8d3564c-8002-11e6-9de2-ef28e68a7e8c.JPG)
_Freedom Plaza, source: Wikipedia_

**Here is the breakdown of cities we are going to concentrate map:**
#### Major roads alignment as per Mapbox imagery

| City name | Task link | Start date | End date | kms of road | # persons worked |
| --- | --- | --- | --- | --- | --- |
| Taipei & New Taipei | [#26](http://tasks.mapbox.com/project/26) | 21/09/2016 | 21/22/2016 | 694 | 12 |
| Taoyuan City | [#27](http://tasks.mapbox.com/project/27) | 22/09/2016 | 22/09/2016 | 434 | 16 |
| Hsinchu | [#28](http://tasks.mapbox.com/project/28) | 22/09/2016 | 22/09/2016 | 341 | 8 |
| Taichung | [#29](http://tasks.mapbox.com/project/29) | 22/09/2016 | 23/09/2016 | 1187 | 19 |
| Yilan | [#30](http://tasks.mapbox.com/project/30) | 23/09/2016 | 23/09/2016 | 507 | 17 |
| Tainan | [#31](http://tasks.mapbox.com/project/31) | 23/09/2016 | 23/09/2016 | 860 | 9 |
#### Minor roads alignment as per Mapbox imagery

| City name | Task link | Start date | End date | kms of road | # persons worked |
| --- | --- | --- | --- | --- | --- |
| Taipei & New Taipei | [#32](http://tasks.mapbox.com/project/32) | 27/09/2016 | 27/22/2016 | 832 | 10 |
| Taoyuan City | [#34](http://tasks.mapbox.com/project/34) | 27/09/2016 | 28/09/2016 | 1283 | 10 |
| Hsinchu | [#35](http://tasks.mapbox.com/project/35) | 28/09/2016 | 28/09/2016 | 517 | 10 |
| Taichung | [#40](http://tasks.mapbox.com/project/40) | 28/09/2016 | 29/09/2016 | 3640 | 20 |
| Tainan | [#41](http://tasks.mapbox.com/project/41) | 29/09/2016 | 29/09/2016 | 2253 | 20 |
| Yilan | [#42](http://tasks.mapbox.com/project/42) | 29/09/2016 | 29/09/2016 | 1843 | 10 |
#### Major and Minor roads alignment as per Mapbox imagery

| City name | Task link | Start date | End date | kms of road | # persons worked |
| --- | --- | --- | --- | --- | --- |
| Kaohsiung | [#43](http://tasks.mapbox.com/project/43) | 29/09/2016 | 30/09/2016 | 4097 | 20 |
| Taipei, New Taipei & Taoyuan Outskirts | [#44](http://tasks.mapbox.com/project/44) | 30/09/2016 | 03/10/2016 | 2566 | 18 |
| Taichung Outskirts | [#39](http://tasks.mapbox.com/project/45) | 03/10/2016 | 03/10/2016 |  |  |
| Hsinchu Outskirts | [#45](http://tasks.mapbox.com/project/39) | 03/10/2016 | 03/10/2016 |  |  |
| Tainan Outskirts | [#46](http://tasks.mapbox.com/project/46) | 03/10/2016 | 04/10/2016 |  |  |
| Yilan Outskirts | [#47](http://tasks.mapbox.com/project/47) | 04/10/2016 |  |  |  |
#### Major and minor road alignment as per bing imagery

| City name | Task link | Start date | End date | kms of road | # persons worked |
| --- | --- | --- | --- | --- | --- |
| Chiayi County - 1 | [#48](http://tasks.mapbox.com/project/48#) | 04/10/2016 |  |  |  |
| Chiayi County - 2 | [#49](http://tasks.mapbox.com/project/49#) | 04/10/2016 |  |  |  |
| North Changhua County | [#50](http://tasks.mapbox.com/project/50#) |  |  |  |  |
| South Changhua County | [#51](http://tasks.mapbox.com/project/51#) |  |  |  |  |
| Chiayi County - 3 | [#52](http://tasks.mapbox.com/project/52#) |  |  |  |  |
| Taipei north | [#53](http://tasks.mapbox.com/project/53#) |  |  |  |  |
| Xinwu District | [#54](http://tasks.mapbox.com/project/54#) |  |  |  |  |
| South Fengling Township | [#55](http://tasks.mapbox.com/project/55#) |  |  |  |  |
| Hualien County | [#56](http://tasks.mapbox.com/project/56#) |  |  |  |  |
| South of Taiwan | [#57](http://tasks.mapbox.com/project/56#) |  |  |  |  |

**Note:** Please do follow the workflow mentioned in the tasking manager for👆  tasks

**Team:** @mapbox/team-data along with the OSM-Taiwan community

**Features to map:** All major roads, from tertiary and above

**Available data sources:** Mapbox imagery and Strava cycling and walking heatmap
### Workflow:
- Load Mapbox Satellite Imagery as base map in JOSM
- Load Strava Heatmap of cycling and walking for reference : _[click this link to load in JOSM](http://127.0.0.1:8111/imagery?title=Strava&type=tms&url=tms%5B22%5D:http://globalheat.strava.com/tiles/both/color2/{zoom}/{x}/{y}.png)_
- Load the selected block of tasking manager in JOSM and add/realign and merge highways as per Mapbox satellite imagery
- Use inverse filter `highway=motorway|trunk|primary|secondary|*link|motorway_link|tertiary` for showing major roads
- Use inverse filter `highway=residential|unclassified|service` for showing minor roads
- Use inverse filter `highway=motorway|trunk|primary|secondary|motorway_link|*_junction|tertiary|trunk_link|residential|unclassified|service` for showing all types of road
- Split the block into smaller blocks if the data is too huge 

_Changeset comment_ - `Add/realign streets #https://github.com/mapbox/mapping/issues/233`

_Changeset source_ - `Mapbox Satellite, Strava heat map`

cc: @mapbox/team-data

**Comment by maning** [comment]
![screen shot 2016-09-21 at 15 18 10](https://cloud.githubusercontent.com/assets/353700/18706203/aaaab132-800e-11e6-8850-e3cadc5c5f9f.png)
*This intersection looks like it should be connected from southeast to northwest*
**Should we connect?**

**Comment by maning** [comment]
@Supaplextw

**Comment by Supaplextw** [comment]
It seems there is a connect road from the lane to the main street, both for cars and pedestrians

**Comment by Supaplextw** [comment]
https://www.openstreetmap.org/#map=18/25.032661/121.495897
@maning I'm not sure which intersection do you mean. Could you use a mark to pin down?

**Comment by maning** [comment]
@Supaplextw This one: http://osm.org/go/5i1pZoyjO--?m=

**Comment by Supaplextw** [comment]
@maning Ya. A connect road and the entrance to the parking lot for the commercial building.

**Comment by maning** [comment]
> Ya. A connect road and the entrance to the parking lot for the commercial building. 

@Supaplextw Thanks! Connected now.

**Comment by jothirnadh** [comment]
We completed 64% 🚀  of the [Taipei task](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/26) in 3 hours with a team of 12 mappers, which implies each person is completing 5 blocks per hour on an average 💥 . There is a total of 1158 blocks in Taiwan tasks yet to be mapped. As per estimated stats with current workflow, we need 3 days for mapping and 3 more days for validation with 20 active mappers (including Peru). 

- [ ] Work on providing regular feedback to the OSM-Taiwan community through this ticket
- [ ] We are mapping only tertiary and above roads for now, but in future, we can think of other tasks for mapping/aligning minor roads as well
- [ ] There are three more cities in the south of Taiwan which we will hit after the Satellite imagery is updated.

cc: @mapbox/team-data

**Comment by maning** [comment]
![screen shot 2016-09-22 at 14 45 05](https://cloud.githubusercontent.com/assets/353700/18742668/2d8152d6-80d3-11e6-85b3-6c30d4a52c1e.png)

These two sections of a motorway overlapped, they have different layer tags, so I assume they are on different levels/height from the ground.  @Supaplextw Can you confirm?

Way ID below:

* https://www.openstreetmap.org/way/229193806/
* https://www.openstreetmap.org/way/312878259

**Comment by Supaplextw** [comment]
Ya, it's a overlap expressway in the city.

**Comment by nikhilprabhakar** [comment]
![layers](https://cloud.githubusercontent.com/assets/11845908/18743712/ee1a26e0-80d7-11e6-8c6b-5bb7d6734d68.gif)


Only one road is traced for `layer=3` and `layer=2` in the above case. But is separately drawn after the intersection. What should I do in the case? Should I trace a separate road or is there a particular reason why only one road is trace? Can you please help @Supaplextw?

 - Way ID: - https://www.openstreetmap.org/#map=19/25.06097889283218/121.5885193048383

**Comment by maning** [comment]
![](http://www.inautonews.com/wp-content/uploads/2011/08/201315-original1-m5ehi.jpg)
*OMG! Taiwan!*

Please watch out for the layer tags.

**Comment by maning** [comment]
![screen shot 2016-09-22 at 15 48 00](https://cloud.githubusercontent.com/assets/353700/18744524/0827ffea-80dc-11e6-8fef-8921e82b640f.png)
location: http://osm.org/go/5i1pdRfkQ--?m=

The tertiary looks like a dual-carriageway and should not connect to the single traffic signal base on imagery.

**Comment by maning** [comment]
To verify layer for complex interchanges, use the `Layer checker` paint style in JOSM.
**View > Map Paint Styles > Map pain preferences**.  

![screen shot 2016-09-22 at 16 03 24](https://cloud.githubusercontent.com/assets/353700/18745003/1f1eccf4-80de-11e6-984f-69f0b220ae8d.png)

**Comment by abhisheksaikia** [comment]
@Supaplextw Is this an elevated U-turn or an error. Can you please go through it? Link: https://www.openstreetmap.org/way/364187901/history#map=19/25.02317/121.27970

![screen shot 2016-09-22 at 4 56 39 pm](https://cloud.githubusercontent.com/assets/13744156/18746530/93c47f8e-80e5-11e6-9173-42d9b71bf22a.png)

**Comment by jothirnadh** [comment]
We completed [Taipei mapping task](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/26) and added/aligned **694 Kms** of roads all together in 2 days 💥 . This is a huge lift off road network improvement in Taiwan and is never possible without the support of OSM-Taiwan community 👏 . It would be great if the community can help us in validating the task and give us consistent feedback to improve present workflow.

### Here is the breakdown of Taipei mapping effort:
* Total Kms of roads improved : **694 Kms**
* Total time spent on this task : **5hrs**
* Total mappers involved : **12**

If we continue mapping with such certainty we can even add/align minor roads (residential, unclassified and service roads) as a secondary round of mapping phase in Taiwan.
 
### Next actions:
- [ ] We started off mapping in [New Taipei](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/27) which is expected to complete by tomorrow
- [ ] Post the stats of New Taipei task once it is done 🚀 
- [ ] @mapbox/peru-office, it will be great if you can join us in the upcoming tasks to hit this much faster 😉 

cc: @lxbarth @srividyacb @maning @jacquestardie

**Comment by bsrinivasa** [comment]
![screen shot 2016-09-22 at 7 38 00 pm](https://cloud.githubusercontent.com/assets/17143802/18751358/5ff1e8c4-80fc-11e6-807e-25a9b8b7d4bb.png)

* Way Id - https://www.openstreetmap.org/way/440544583

This way is running over a river and it has no `layer` tag or `bridge`tag.  Can someone with local knowledge check if the data is correct around this road?

Also, the roads at the interchange seems to have no proper `layer` tags tagged for the ways. 

The `bridge=viaduct` is tagged as `layer=4` at one side and `layer=5` for the opposite side which runs for a long distance. 

![screen shot 2016-09-22 at 7 50 58 pm](https://cloud.githubusercontent.com/assets/17143802/18751806/f8bd68ac-80fd-11e6-80da-8c8bb2242caf.png)
*Highways at Interchange with no proper `layer=*` tags*

It would be great if anyone from the community can help us on these issues. 


Here are the way-ids.
* https://www.openstreetmap.org/way/291356413
*  https://www.openstreetmap.org/way/311562578


cc @Supaplextw

**Comment by ridixcr** [comment]
### help me please 

![question](https://cloud.githubusercontent.com/assets/1646072/18763109/7bc29f56-80d1-11e6-8925-394f226a70fb.png)


overlaping between the way and buildings when aligning the way according `Strava` 

https://www.openstreetmap.org/#map=18/24.14457/120.677613
http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/29#task/338

cc. @jothirnadh @maning

**Comment by assanges** [comment]
Hi, it's me assanges. And I guess some of you all had heard of me already.

![image](https://cloud.githubusercontent.com/assets/4113063/18766143/5735c3f0-814b-11e6-9048-ec9cc50b58c1.png)
And I still active on this specific area which I lived hereby, of which I noticed Mapbox analyst had edited again. haha Please kindly take my Strava activity (https://www.strava.com/activities/714223008) or GPX uploaded (https://www.openstreetmap.org/user/assanges/traces/2261480) for reference, and attached official file as well. Many thanks!

For http://www.openstreetmap.org/note/722825
![image](https://cloud.githubusercontent.com/assets/4113063/18766284/0dba127a-814c-11e6-9ef7-87b8c0ea8208.png)

For
http://www.openstreetmap.org/note/722830
http://www.openstreetmap.org/note/722834

![image](https://cloud.githubusercontent.com/assets/4113063/18766374/894da4f6-814c-11e6-89a9-28f7f81c746b.png)

**Comment by abhisheksaikia** [comment]
### Taiwan update and next actions:

The @mapbox/team-data  have almost completed aligning major roads in these 4 cities: Taipei, New Taipei, Hsinchu and Taichung ahead of schedule 💥 .

Next actions in Taiwan realignment:

* [ ] Align major roads in next two cities where satellite imagery is already rendered: **Tainan** and **Yilan**; @jothirnadh to create the tasks and update this ticket.
* [ ] Create a task for the last city: **Kaohsiung City** once the satellite imagery is rendered.
* [ ] Start validating these tasks once major road realignment in all the 7 cities is completed.
* [ ] Start off with aligning minor roads (once major road realignment is over in all the 7 cities): start with Taipei and evaluate based on the time required to realign minor roads in taipei.

cc @shvrm @maning @srividyacb

**Comment by jothirnadh** [comment]
> Align major roads in next two cities where satellite imagery is already rendered: Tainan and Yilan; @jothirnadh to create the tasks and update this ticket.

The tasks for [Tainan](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/31) and [Yilan](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/30) are created. Jump into these tasks once we are done with [Taichung](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/29)

cc: @mapbox/team-data

**Comment by jothirnadh** [comment]
> overlapping between the way and buildings when aligning the way according to Strava and Mapbox Satellite 

@ridixcr if the buildings are overlapping with the imagery, make sure that the buildings are aligned to imagery as well. It is documented in the [workflow](https://github.com/mapbox/mapping/issues/233#issue-178282328) as well.

**Comment by jothirnadh** [comment]
@Yurasi I came across a road which you aligned as per Strava heat map
![screen shot 2016-09-23 at 12 02 43 pm](https://cloud.githubusercontent.com/assets/8401827/18776291/c2305bea-8185-11e6-8523-ebd03df45476.png)
[Link](https://www.openstreetmap.org/way/241901886/history)

The alignment is supposed to be made as per the Mapbox satellite imagery but not Strava heat map. Just use Strava heat map for support when you are not sure of the road alignment/existence. The reason for this is Strava heatmap consists of the GPS traces from walking and cycling profiles, which lie on the sides of the roads most of the time. In the above case, there is a separate cycling track next to the tertiary road where Strava activity is more. Please make a note of this issue in further mapping @mapbox/team-data .

**Comment by bsrinivasa** [comment]
Hello @assanges ,

Thank you for validating our edits during adding/aligning the roads in Taiwan. We used the existing Mapbox imagery for realigning roads in that area. We tried [editing it back](https://www.openstreetmap.org/changeset/42321756) but seems like it's hard to track things on the ground, as the imagery is a bit old.  It would be great if you can revert/make changes of our edits with your local knowledge so that the map is up to date. Thank you for providing the various sources. Will keep these issues in mind going forward. It would be awesome to have your feedback and participation in helping us validate the task https://github.com/mapbox/mapping/issues/233.

**Comment by maanyaumashaanker** [comment]
![screen shot 2016-09-23 at 4 03 13 pm](https://cloud.githubusercontent.com/assets/19161907/18783083/78cdf800-81a7-11e6-8038-e537035727fc.png)

Could someone please confirm how far does this road having https://www.openstreetmap.org/way/279154065 as its way Id continue?

cc @Supaplextw

**Comment by assanges** [comment]
Hi @bsrinivasa !
Thanks for your fast reply! And yeah, I agreed that it's hard to revert after many edits without field research, and I shall take this from here, a big big thanks again for you and your teammates' effort!

**Comment by kagami3421** [comment]
Hello @maanyaumashaanker .
The secondary road ends tertiary road refenences Y4(宜4) , therefore the road in your screenshot should be unclassified road.

**Comment by abhisheksaikia** [comment]
We completed adding/aligning more than **4000 Kms** of major roads all together in 3 days in [6 cities in Taiwan](https://github.com/mapbox/mapping/issues/233#issue-178282328)💥 . There has also been continuous validation and help from the Taiwan community while we have mapped. A massive thanks to the community for their efforts from @mapbox/team-data  🚀 . It would be great to continue working with the community in the validation work in these 6 cities.

### Here is the breakdown of Taipei mapping effort:
* Total Kms of roads improved : **4023 Kms**
* Total time spent on this task : **3 days**
* Total mappers involved : **19**

### Next actions:
* [ ] Start validating the 6 tasks along with the community.
* [ ] Create a new task for **Kaohsiung City** once the satellite imagery is rendered.
* [ ] Start aligning minor roads in [Taipei](http://cfn-tasking-manager-staging-vpc-387856624.us-east-1.elb.amazonaws.com/project/32)



cc: @Supaplextw @assanges @lxbarth @jacquestardie

**Comment by abhisheksaikia** [comment]
Diary post on Taiwan mapping progress is out: http://www.openstreetmap.org/user/saikabhi/diary/39543

**Comment by nikhilprabhakar** [comment]
@mapbox/team-data, While [validating the task in Yilan](http://tasks.mapbox.com/project/30#task/85), I came across some issues in the data.
 
##### Case 1:
![issue1](https://cloud.githubusercontent.com/assets/11845908/18864224/eb15f724-84b2-11e6-93fc-68435cda881e.jpg)

*Always make sure the road is aligned to the centre of the road in satellite imagery*

##### Case 2:
![issue2](https://cloud.githubusercontent.com/assets/11845908/18864050/0fb13d4c-84b2-11e6-942c-0b9892146a56.jpg)
*Make sure the `motorway_junctions` are aligned to where the road splits*

##### Case 3:
![aligningjunctions](https://cloud.githubusercontent.com/assets/11845908/18864107/666638ea-84b2-11e6-8ae6-908cb31de40b.gif)
*While realigning roads in junctions, always make sure the road that crossings the junction is also realigned in a straight line.*

Let's make sure we are checking all these above cases and updating appropriately while validating the data.

**Comment by abhisheksaikia** [comment]
As a reminder to everyone already validating, or jumping in on validation process of major roads in Taiwan - 

* we've completed validating 5 cities, and right now validation is underway in Tainan
* remember that we're using Mapbox imagery to align roads - make sure the same imagery is used when validation as well
* when using Strava (cycling or walking traces), often the traces fall to the side of the road. Only use Strava as a guidance source, rather than the primary source for realignment

![misalignment1](https://cloud.githubusercontent.com/assets/13744156/18873319/69b98fcc-84dc-11e6-8fe1-bc5b17d16105.gif)
*Bing vs. Mapbox Imagery misalignment example*

cc @mapbox/team-data

**Comment by abhisheksaikia** [comment]
There are cases of satellite imagery distortion in `Mapbox`. Distortion can be identified by sudden curves in long stretches of highways which are generally straight. Let's keep an eye out for such issues while aligning and validating blocks. In case, `Mapbox` appears distorted double check with `Bing` and `Strava` traces to verify the distortion.

![mapbox_satellite_distortion333](https://cloud.githubusercontent.com/assets/13744156/18901192/8e19d522-8566-11e6-89d5-5775d95135d9.gif)

cc @mapbox/team-data @jacquestardie

**Comment by chtnha** [comment]
Many [highways are overlapping on lot of buildings](https://www.openstreetmap.org/#map=17/24.13941/120.68419). The buildings doesn't match to any imageries (Mapbox/Bing). The user [CYUT10128068](https://www.openstreetmap.org/user/CYUT10128068) added these buildings 11 months ago. 

![screen shot 2016-09-28 at 4 40 51 pm](https://cloud.githubusercontent.com/assets/6770741/18912113/60c18f42-859e-11e6-9c10-ceafc903ba49.png)


#### The testing buildings:

Here in this junction, the buildings are serially added. 

![screen shot 2016-09-28 at 5 14 37 pm](https://cloud.githubusercontent.com/assets/6770741/18912351/904486d8-859f-11e6-85fe-16941f1f1006.png)

![screen shot 2016-09-28 at 5 14 23 pm](https://cloud.githubusercontent.com/assets/6770741/18912349/903e64ec-859f-11e6-981d-bc9ab79cbf6a.png)


The actual buildings looks completely different. (*Look right side*) 

![screen shot 2016-09-28 at 5 14 53 pm](https://cloud.githubusercontent.com/assets/6770741/18912350/903ef98e-859f-11e6-833b-3a449e9e5eb6.png)

How we are moving with this kind of overlappings? and ensure the quality of the data. 

@assanges @Supaplextw @mapbox/team-data

**Comment by kagami3421** [comment]
Hello @chtnha .
The account is shared by some students from CYUT university(朝陽科技大學).
Their doing also preplexs us. Such as drawing buildings without considering real situation.

Now the best treatment jsut ingore them. Seriously , fix the overlapping will annoy and increase your work.

**Comment by Supaplextw** [comment]
The editing from CYUT ignore the real building situation. We're thinking about mass deletion of there building due to lack of response and fixing.

**Comment by assanges** [comment]
I agreee with kagami3421. These improper edit shouldn't set you back in efforts on alignment, leave them to we the local community to fix it.

**Comment by Rub21** [comment]
Hi Everyone ,
 Usually in alignment we should move many  nodes and we need to be sure where we worked and where missing realignment to check it I am using this  [JOSM style](https://gist.githubusercontent.com/Rub21/9f0aedb1f9b604843fc677d63939f0e5/raw/50881225329a8d04354b5345b90183e74b75a4b7/modifiedNodes.mapcss) which help me to check out where I modified the  objects, the differences between the existing  `modified objects` style  is that this works only for modified nodes, and is more clear to me.
![image](https://cloud.githubusercontent.com/assets/1152236/18956639/bfc4bbc6-8621-11e6-8edf-de05e3ac2c79.png)

**Comment by littlebtc** [comment]
FYI, I am adjusting the Wenxin Road (文心路) in Taichung again, as the train viaduct is building in the center of the road and the imagery is not updated for that.

**Comment by littlebtc** [comment]
In the imagery, the Wenxin Road near Wenxin Park:

![Imgur](http://i.imgur.com/SbZRTGr.png)

The actual street view now after the viaduct is completed:

![Imgur](http://i.imgur.com/Nrb9s8P.jpg)

So I have to re-align the road and the railway viaduct to match the current status.

**Comment by jothirnadh** [comment]
We completed aligning more than **10,368 Kms** of minor roads all together in 3 days for [6 highly populated cities of Taiwan](https://github.com/mapbox/mapping/issues/233)💥. Massive thanks to the OSM-Taiwan community for their support from @mapbox/team-data 🚀. It would be great to continue working with the community in the validation work in these 7 cities. The team will move ahead with aligning minor and major roads in the outskirts of these cities.

### Here is the breakdown of Taiwan mapping effort:
* Total Kms of roads improved : **10,368 Kms**
* Total time spent on this task : **3 days**
* Total mappers involved : **20**

### Next actions:
* [ ] Take help from the community to validate the tasks.
* [ ] Complete the road alignment task in outskirts of top 7 cities ([Taichung](http://tasks.mapbox.com/project/39), [Yilan](http://tasks.mapbox.com/project/47), [Tainan](http://tasks.mapbox.com/project/46), [Hsinchu](http://tasks.mapbox.com/project/45) still pending)
* [ ] Create new tasks for areas outside newly rendered satellite imagery for road realignment.

cc: @Supaplextw @assanges @kagami3421 @lxbarth

**Comment by Yurasi** [comment]
Hi team! I have a question, must we change the `highway=*` tag? i.e. We find some street with `highway=service` tag, but they should be clearly `highway=residential`.

![seleccion_002](https://cloud.githubusercontent.com/assets/12261974/19042506/f3c47c80-8951-11e6-96bd-5519d04aa26a.jpg)

But also there are some of them tagged as `highway=road` (which are motorable), and I'm wondering how much important is change them because of the navigability of the city through all the motorable roads.

![seleccion_001](https://cloud.githubusercontent.com/assets/12261974/19041956/ca256634-894f-11e6-9c5a-d5311042fc8c.jpg)

cc: @Supaplextw @assanges @kagami3421

**Comment by maning** [comment]
@Yurasi, lets focus on realigning for now.  The community can then review the road class.

**Comment by jothirnadh** [comment]
Great work @mapbox/team-data and OSM Taiwan community 👏 . We are almost done with aligning roads in the top [7 cities of Taiwan](https://github.com/mapbox/mapping/issues/233#issue-178282328) where new Mapbox imagery was added. Let's keep up the zest and continue the mapping quest to the other parts of Taiwan as well 🚀 . 

In the other parts of Taiwan, the Mapbox imagery is limited to zoom 17. So, from now on let's stick to Bing imagery for mapping. Will post more updates on the upcoming tasks. Let's start off with [Chiayi County - 1](http://tasks.mapbox.com/project/48#) task for now.

Thanks for coordinating with us so far OSM Taiwan community 🙇 , it would be great if the community members can validate the completed tasks and give us feedback to improve our workflow.

**Comment by Andygol** [comment]
In some places Mapbox and Bing imagery is shifted. You may align them using Strava heatmap as reference. 

I use 👇 this values for http://tasks.mapbox.com/project/47#task/73 and http://tasks.mapbox.com/project/47#task/31
Bing offset - `-0.00; 4.33`
Mapbox offset - `-0.15; 5.82`

![oct-04-2016 17-51-11](https://cloud.githubusercontent.com/assets/369696/19079514/8a1a65d6-8a5c-11e6-8c77-38b15bff6c9d.gif)

**Comment by kagami3421** [comment]
Hello @Yurasi .
Your change is correct and helps me a lot. But changing road class is our task I should do.

Those service road is tagged by me when starting contributing OSM in 2014~2015.
I've considered the Lane (巷) should tagging as service road , but other guys say my tagging schema is wrong. 
Therefore I changed my schema after according to OSM wiki.

**Comment by nikhilprabhakar** [comment]
@mapbox/team-data - In case you find any alignment/offset issues in the Satellite Imagery that is being used for a task, always cross check the area with Strava heat map for aggregated GPS traces before adding/modifying the data. Below are the two Strava heat map [TMS layers that can be added to JOSM](https://josm.openstreetmap.de/wiki/Help/Preferences/Imagery#AddcustomTMSimagery):

 - Strava Cycling - `tms[16]:http://globalheat.strava.com/tiles/cycling/color3/{zoom}/{x}/{y}.png)`

- The Strava Global heat map  - `tms[3,17]:http://globalheat.strava.com/tiles/both/color2/{zoom}/{x}/{y}.png`

**Comment by maning** [comment]
![screen shot 2016-10-11 at 16 38 35](https://cloud.githubusercontent.com/assets/353700/19268315/715848e0-8fd1-11e6-9413-585ba63558f9.png)
*Taiwan Bing imagery offsets at http://offsets.textual.ru/map*

Currently, we provide imagery offset adjustment through the specific TM when we use Bing.  There is a [db for storing and sharing imagery offset](http://wiki.openstreetmap.org/wiki/Imagery_Offset_Database) that is integrated in JOSM.  **Should we update the Imagery Offset DB so the community can use it too?**

@Supaplextw @assanges @kagami3421 @littlebtc Is this something that the OSM-TW community use for their editing with Bing?


cc/ @Andygol @jothirnadh @abhisheksaikia @srividyacb @planemad

**Comment by Andygol** [comment]
@maning I ❤️  ☝️ 

as I know this offset db is used in **Vespucci** - https://wiki.openstreetmap.org/wiki/Vespucci

**Comment by littlebtc** [comment]
I was using the JOSM offset plugin when dealing with Bing.

**Comment by luiswalter** [comment]
The way of using the Strava alignment layer and Bing are not very good in this area - http://tasks.mapbox.com/project/71#task/1203
![imagen strava and bing](https://cloud.githubusercontent.com/assets/17984615/19281989/1388678c-8fb2-11e6-8565-a20a3513730a.png)

It was worked with Satellite MapBox which alignment is very good.

![imagen mapbox satellite](https://cloud.githubusercontent.com/assets/17984615/19281999/19026d5c-8fb2-11e6-895b-dcdb6366af35.png)

What are the values to consider higher value to align properly? Bing and Strava o Mapbox satellite

@Rub21 @jothirnadh @maning

**Comment by maning** [comment]
![realign_to_strava](https://cloud.githubusercontent.com/assets/353700/19380910/005a90d8-9217-11e6-95e2-fb7e79d90c68.gif)

For hilly areas and when there is very good Strava traces, always rely on Strava.  The imagery for both Bing and Mapbox is outdated.  The new segment of road was created here (Oct 2015).  See for example [here](https://www.google.co.in/maps/@23.0632995,120.6207003,3a,75y,179.56h,71.64t/data=!3m6!1e1!3m4!1sYh6XH_SD7SJNYwRT3i0NaQ!2e0!7i13312!8i6656?hl=en)

**Comment by srividyacb** [comment]
Such an amazing work everyone 🚀 🎉 👏 

With [42 tasks in Tasking Manager](http://tasks.mapbox.com/?sort_by=priority&direction=asc&search=taiwan) and team of **25 mappers,** we have completed aligning highways (`highway=service` and up) **57,342 kms** of roads in Taiwan which is **48%** of overall Taiwan road network. By doing this alignment we have covered **25,390 square kms - 70%** of the total area of Taiwan. 

Area Covered 🗺  | Data team Edits 👏 
------------ | -------------
![image](https://cloud.githubusercontent.com/assets/3423533/19377366/6cda62e2-9203-11e6-95c3-735065620297.png) | ![](https://cloud.githubusercontent.com/assets/353700/19378862/97ae70ae-920c-11e6-94ed-028e22db79d2.gif)

**Next Actions**

- [ ] Next the team will be working on validating the finished tasks and it will be really helpful if Taiwan community also be part of validation ( @Supaplextw @assanges @kagami3421 ) and provide inputs to the team.

cc @mapbox/team-data

**Comment by jinalfoflia** [comment]
While validating this [block](http://tasks.mapbox.com/project/71#task/199), I came across numerous independent nodes present around the actual way, this happens when we resolve conflicts. While resolving, we tend to take _**other's version**_ over ours and the extra nodes added by us during re-aligning will be left behind. Since we are using this filter `highway=motorway|trunk|primary|secondary|motorway_link|*_junction|tertiary|trunk_link|residential|unclassified|service` these stray nodes will not be highlighted. @mapbox/team-data please look in for such nodes while validating the blocks.

![nodes](https://cloud.githubusercontent.com/assets/4470913/19384078/188ea56e-9225-11e6-9b2a-d369ee550beb.gif)

cc: @maning @planemad @jothirnadh

**Comment by bsrinivasa** [comment]
@mapbox/team-data used both newly rendered Mapbox imagery and Bing imagery (with offsets at few regions) for re-alignment of roads in Taiwan. 

We decided to try out a plugin for JOSM - [Imagery Offset Database](http://wiki.openstreetmap.org/wiki/Imagery_Offset_Database) which stores the offset for the imagery at the location. This [plugin](http://wiki.openstreetmap.org/wiki/Imagery_Offset_Database#JOSM_Plugin) has been widely used by the community for storing various offsets for imagery around the world.

The offsets used by the data team for various [tasks ](http://tasks.mapbox.com/)from tasking manager will be updated in this database. This would help community while mapping/validating in that region using the particular imagery.

We would love to hear any suggestion from community on this and as well if there are any other tools to store and share the offsets used at various regions for specific imagery in Taiwan. 

cc: @maning @Andygol @jothirnadh @krishnanammala l

**Comment by srividyacb** [comment]
Taiwan Wrap-up diary - http://www.openstreetmap.org/user/srividya_c/diary/39839. No next actions. Closing.

**Comment by krishnanammala** [comment]
During Taiwan Road realignment , we came across [overlapping buildings](https://github.com/mapbox/mapping/issues/233#issuecomment-250147540) which were added by CYUT university students. The community members are so active in deleting those buildings as they mentioned in the comments [1](https://github.com/mapbox/mapping/issues/233#issuecomment-250156604), [2](https://github.com/mapbox/mapping/issues/233#issuecomment-250160117), [3](https://github.com/mapbox/mapping/issues/233#issuecomment-250299124).

![screen shot 2016-11-11 at 11 29 29 am](https://cloud.githubusercontent.com/assets/17470597/20205961/05dfb9aa-a803-11e6-89f1-4c80158afc2b.png)

I came to know about this when I am working on [mass deletion filter for Taiwan](http://osmcha.mapbox.com/?bbox=119.96%2C21.51%2C122.48%2C25.45&is_suspect=False&is_whitelisted=True&harmful=False&checked=False&reasons=3) today.  

Taiwan Community Rocks 💥
